src/sql/src/pure.rs

```diff
@@ -813,7 +626,223 @@ pub async fn purify_create_source(
 
     purify_source_format(&*catalog, format, connection, envelope, &connection_context).await?;
 
-    Ok((subsources, stmt))
+    Ok((subsources, Statement::CreateSource(stmt)))
+}
+
+async fn purify_alter_source(
```

Comment:
Some doc-comments would probably help understand what the return type is.

Reply:
This had as much documentation as `purify_create_source` :D Added something